doc/ssh_config/ssh_config.md

```diff
@@ -0,0 +1,214 @@
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+	SSH - secure shell
+	TCP - Transmission Control protocol	
+	
+###  1.4. <a name='Overview'></a>Overview 
+
+We want to allow configuring ssh server global settings. This will feature will include 3 configurations in the first phase, but can be extended easily to include additional configurations.
+
+###  1.5. <a name='Requirements'></a>Requirements
+
+This feature requires a dedicated table in the configuration DB, and enhancements of hostcfg demon, in order to allow modifing the relvant ssh configuration files. In order to override ssh configurations, we need to have write access to ssh config files such as /etc/ssh/sshd_config
+
+###  1.6. <a name='ArchitectureDesign'></a>Architecture Design 
+####  1.6.1. <a name='ConfigModules'></a>Configuration modules
+![ssh_config_flow](ssh_config_flow.png)
+
+We want to enhance configDB to include table for ssh server global configurations. In addition, hostcfg demon will include a dedicated flow in order to modify ssh config files, once ssh server global config table entries are changed.
+
+###  1.7. <a name='High-LevelDesign'></a>High-Level Design 
+
+We want to enable global ssh server configuration in SONIC. In order to do so will touch few areas in the system:
+1. configDB - to include a dedicated table for configurations
+2. hostcfg demon - to update ssh config files once configDB relevant areas are modified (and for this feature, ssh server config table)
+3. OS ssh config files - specific for this stage we are only /etc/ssh/sshd_config is going to be modifed by the hostcfg demon.
+4. OS ssh service - to be restarted after each configuration change.
+
+Note:
+
+The Daemon is running in the host (without container) that matches this feature, because it is basically writing policies on ssh config files from the OS.
+##### Flow diagram
+![ssh_flow_community](ssh_flow_community.png)
+#### 1.7.1 <a name='Flow description'></a>Flow description
+When the feature is enabled, by modifying the DB manually, user will set ssh server policies/configuration (see options below) by modifing CONF_DB in SSH_SERVER_TABLE.
+
+The hostcfgd daemon will be extended to listen to ssh policies/configurations from SSH_SERVER table, parse the inputs and set the new policies to ssh config files, and update ssh server afterwards.
+
+
+#### 1.7.2 <a name='ssh server policies'></a>ssh server policies
+
+We want to enable configuring the following policies, with default values are taken from OS (Debian):
+| Policy         |      Action                                                               | Param values            | Default OS value |
+|----------------|---------------------------------------------------------------------------|-------------------------|------------------|
+| authentication retries |     Number of attempts to try to log in   before rejecting the session    |     3-100               |     6            |

```

Comment:
>retries [](http://example.com/codeflow?start=17&length=7)

tries will be more accurate.

Reply:
I don't think it matters, will leave it that way if you approve